docs/workshops/advanced/circuit-workflow.md

```diff
@@ -0,0 +1,154 @@
+
+Try to do this on your own, however, if you get stuck, here is a working implementation:
+
+??? example
+    ```python
+    @step("Modify")
+    def modify(subscription: Circuit, under_maintenance: bool) -> State:
+        logger.debug(
+            "Changing circuit maintenance state",
+            subscription=subscription,
+            type=type(subscription),
+            old_value=subscription.circuit.under_maintenance,
+            new_value=under_maintenance,
+        )
+        # Set the subscription under_maintenance value to what the user input.
+        subscription.circuit.under_maintenance = under_maintenance
+
+        return {"subscription": subscription}
+    ```
+
+Once you have made your implementation, save the file, and the orchestrator backend will hot-reload. Run a new modify workflow on a circuit. Once you've run the workflow, you should now be able to go into the subscription data tab and see the new value for the under_maintenance field. If you get really fancy with it, click on the `Delta` button to see the exact changes to the subscription. If everything was done right, you should see the following change to the subscription:
+
+```json
+{
+  "circuit": {
+    "under_maintenance": false
+  }
+}
+```
+
+## Exercise 2: Updating the Configuration Based Off the Domain Model
+
+Now that we are actually updating the orchestrator's view of things, we need to go ahead and make sure that our intent is actually applied to the network! Using our simple copy/paste method of applying network intent, this should be a fairly easy fix. 
+
+First things first, think back to how we are providing the config that we display to the user in the `provide_config_to_user()` step function. We are simply pulling data out of the subscription domain model and the populating strings with the values. Assuming that the CLI we are using is idempotent, we can simply re-apply all that config with the new values and the device config will be updated.
+
+!!! warning
+
+    Since this is just an example workshop, this will work for us. In prodcution, you will want to use a much more robust configuration mediation engine, especially in a multi-vendor network, however, for the purposes of this workshop, copy/pasting config will suffice.
+
+With this in mind, all we really need to do is take the `provide_config_to_user()` step from shared, import it into our modify workflow/add it to the steplist and be off to the races! 
+
+Go ahead and try to do this on your own, however, if you get stuck, here is a working implementation:
+
+??? example
+
+    First, go ahead and import the step from the shared file, like so:
+    ```python
+    from workflows.circuit.shared import provide_config_to_user
+    ```
+
+    Now go ahead and add that to the steplist so that your steplist looks like this:
+
+    ```python
+    def modify_circuit() -> StepList:
```

Comment:
I guess we introduced the `@modify_workflow` decorator so this might work a bit differently now.